internal/kibana/role.go

```diff
@@ -0,0 +1,506 @@
+			if v, ok := userElasitcConfig["cluster"]; ok {
+				definedCluster := v.(*schema.Set)
+				cls := make([]string, definedCluster.Len())
+				for i, cl := range definedCluster.List() {
+					cls[i] = cl.(string)
+				}
+				kibanaRole.Elasticsearch.Cluster = cls
+
+				if v, ok := userElasitcConfig["indices"]; ok {
+					definedIndices := v.(*schema.Set)
+					indices := make([]kbapi.KibanaRoleElasticsearchIndice, definedIndices.Len())
+					for i, idx := range definedIndices.List() {
+						index := idx.(map[string]interface{})
+
+						definedNames := index["names"].(*schema.Set)
+						names := make([]string, definedNames.Len())
+						for i, name := range definedNames.List() {
+							names[i] = name.(string)
+						}
+						definedPrivileges := index["privileges"].(*schema.Set)
+						privileges := make([]string, definedPrivileges.Len())
+						for i, pr := range definedPrivileges.List() {
+							privileges[i] = pr.(string)
+						}
+
+						newIndex := kbapi.KibanaRoleElasticsearchIndice{
+							Names:      names,
+							Privileges: privileges,
+						}
+
+						if query := index["query"].(string); query != "" {
+							newIndex.Query = &query
+						}
+						if fieldSec := index["field_security"].([]interface{}); len(fieldSec) > 0 {
+							fieldSecurity := map[string]interface{}{}
+							// there must be only 1 entry
+							definedFieldSec := fieldSec[0].(map[string]interface{})
+
+							// grants
+							if gr := definedFieldSec["grant"].(*schema.Set); gr != nil {
+								grants := make([]string, gr.Len())
+								for i, grant := range gr.List() {
+									grants[i] = grant.(string)
+								}
+								fieldSecurity["grant"] = grants
+							}
+							// except
+							if exp := definedFieldSec["except"].(*schema.Set); exp != nil {
+								excepts := make([]string, exp.Len())
+								for i, except := range exp.List() {
+									excepts[i] = except.(string)
+								}
+								fieldSecurity["except"] = excepts
+							}
+							newIndex.FieldSecurity = fieldSecurity
+						}
+
+						indices[i] = newIndex
+					}
+					kibanaRole.Elasticsearch.Indices = indices
+				}
+
+				if v, ok := userElasitcConfig["run_as"]; ok {
+					definedRuns := v.(*schema.Set)
+					runs := make([]string, definedRuns.Len())
+					for i, run := range definedRuns.List() {
+						runs[i] = run.(string)
+					}
+					kibanaRole.Elasticsearch.RunAs = runs
+				}
+			}
+		}
+	}
+
+	if v, ok := d.GetOk("metadata"); ok {
+		metadata := make(map[string]interface{})
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&metadata); err != nil {
+			return diag.FromErr(err)
+		}
+		kibanaRole.Metadata = metadata
+	}
+
+	roleManageResponse, err := kibana.KibanaRoleManagement.CreateOrUpdate(&kibanaRole)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, diags := client.ID(ctx, roleManageResponse.Name)
+	if diags.HasError() {
+		return diags
+	}
+
+	d.SetId(id.String())
```

Comment:
I know you've just copied it, but this creates a dependency on Elasticsearch being configured when managing a Kibana resource. Let's just use the role name as the ID
```suggestion
	d.SetId(roleManageResponse.Name)
```